src/py/flwr/server/utils/validator.py

```diff
@@ -64,21 +64,8 @@ def validate_task_ins_or_res(tasks_ins_res: Union[TaskIns, TaskRes]) -> List[str
             validation_errors.append("non-anonymous consumer MUST provide a `node_id`")
 
         # Content check
-        has_fields = {
-            "sa": tasks_ins_res.task.HasField("sa"),
-            "legacy_server_message": tasks_ins_res.task.HasField(
-                "legacy_server_message"
-            ),
-        }
-        if not (has_fields["sa"] or has_fields["legacy_server_message"]):
-            err_msg = ", ".join([f"`{field}`" for field in has_fields])
-            validation_errors.append(
-                f"`task` in `TaskIns` must set at least one of fields {{{err_msg}}}"
-            )
-        if has_fields[
-            "legacy_server_message"
-        ] and not tasks_ins_res.task.legacy_server_message.HasField("msg"):
-            validation_errors.append("`legacy_server_message` does not set field `msg`")
+        if not tasks_ins_res.task.HasField("recordset"):
+            validation_errors.append("`recordset` MUST be set")
```

Comment:
Should we add `task_type` as mandatory as well?

Reply:
I think checking `task_type` makes sense. Will update this.